### PR TITLE
Revise Array, ArrayBuffer, BigInt, DataView builtins

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: third_party/yarr/Vector.h [hunk 1]
```diff
@@ -210,6 +210,187 @@ class Vector {
 
 };
 
+template <typename T>
+class GCVector {
+public:
+    using iterator = T*;
+    using const_iterator = const T*;
+    Escargot::Vector<T, GCUtil::gc_malloc_allocator<T>> impl;
```
Comment (clover2123) [inline third_party/yarr/Vector.h:218]: minor) what about defining `impl` as a private member, not public?

Comment (ksh8281) [reply third_party/yarr/Vector.h:218]: Good idea!

Comment (ksh8281) [reply third_party/yarr/Vector.h]: I've updated pr

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**